### DISCUSSION TITLE
hash user's ip for key generation

### DISCUSSION THREAD

**Discussion by ItsANameToo** [danharrin/livewire-rate-limiting#12] [Ideas]
Could the user's IP address that's being used to generate a key be adjusted to make use of a hash instead? The line in question is https://github.com/danharrin/livewire-rate-limiting/blob/main/src/WithRateLimiting.php#L23

Having it as a hash of the IP instead would bring it in line with the way Laravel generates a key for throttling (see https://github.com/laravel/framework/blob/9.x/src/Illuminate/Routing/Middleware/ThrottleRequests.php#L172) and will also avoid using user-identifiable data when not necessary.

**Comment by danharrin** [comment]
Surely a hash will be calculated differently every time, hence the key will change and the rate limit wont work? Have you tested it?

**Comment by ItsANameToo** [comment]
when using `sha1` it will not result in different hashes for the same input, unless you add a random salt to it. You can see it with tinker vs an online website:
```
> sha1('string')
= "ecb252044b5ea0f679ee78ec1a12904739e2904d"
```

http://www.sha1-online.com/:
<img width="661" alt="image" src="https://user-images.githubusercontent.com/35610748/202744269-12c0132c-78a6-4443-b2db-c9cbb13cd9c1.png">

**Reply by danharrin** [reply]
Fair enough. Can you PR then?

**Reply by ItsANameToo** [reply]
Sure, I'll get to that on monday!

**Reply by ItsANameToo** [reply]
https://github.com/danharrin/livewire-rate-limiting/pull/13
